### ISSUE TITLE
Change installation scripting to use native Lua commands

### ISSUE THREAD

**Issue by norcalli** [neovim/nvim-lspconfig#92]
I had a thought today that I could provide a function to make this easier which looks like this:

```lua
util.script(function(sh, print)
  sh.curl{"https://..."}
  sh.julia({}, {stdout = function()...end; onexit=...})
  print("ASLKDJLAKSJD")
end)
```

This would allow us to choose how to implement the backend and abstract it away, for instance by redirecting the output of the commands to a buffer (like we do currently with `util.sh`) and still make it ergonomic to use.

I already have a prototype I use for my own things which has the interface I'm describing above, and I've found it very useful in practice. We could even use luv (with coroutines to synchronize execution) instead of vim.fn.system to execute things to avoid some overhead, but that may also just be overoptimizing and we can return the output each time anyway.

This would also probably be a useful utility to go in neovim core too.

**Comment by teto** [comment]
There are just too many ways to install software that I don't believe it's worth creating our own package manager. It depends on the platform, the ecosystem etc.

**Comment by justinmk** [comment]
@teto Although I agree that installation in general is not a business we want to be in, that's not what this issue is proposing. 

We already have some best-effort attempts in this plugin to install various things. And this issue AFAIU is proposing to make some util methods to do that instead of writing a bunch of shell scripts everywhere. And that's definitely a good idea.

Regarding the general idea of installing stuff: I think we can manage it as long as we consistently refuse to support it, i.e. it's best-effort. Similar to `:checkhealth`, which only tries its best, and leaves the rest up to the user.

**Comment by mjlbach** [comment]
Closed by #498
